Lesson17/Lesson17/gokce/script.js

```diff
@@ -0,0 +1,217 @@
+for (let info in person){
+    console.log(info + ": " + person[info]);
+}
+ 
+
+// 11. Write a for...in loop that sums the values of an object.
+console.log("Exercise 11");
+const obj = { a: 10, b: 20, c: 30, d: 23 };
+let sum = 0;
+for (let numbers in obj){
+    sum += obj[numbers];
+    console.log ("the sum of the numbers are" + sum);
+}
+
+
+// Excercises below are additional (as requested during the lesson)
+
+// 12. Sum of Random Numbers: Write a do-while loop that generates random numbers between 1 and 5 and adds them together until the sum is greater than 20. Print the total number of iterations and the final sum.
+console.log("Exercise 12");
+let totalResult = 0;
+let totalIterations = 0;
+do{
+ let randomNum = Math.floor(Math.random()*5)+1;
+ totalResult += randomNum;
+ console.log(totalResult);
+ totalIterations++;
+}
+while(totalResult<=20);
+console.log("iterations : " + totalIterations);
+console.log("total result : " + totalResult );
+
+
+// 13. Count Even Numbers: Write a for loop that generates and checks 100 random numbers between 1 and 50. Count how many even numbers are generated and print the count at the end.
+console.log("Exercise 13");
+let sumIteration = 0;
+let evenNum =0;
+do{
+let fullNum = Math.floor(Math.random()*50)+1; 
+console.log(fullNum);   
+sumIteration++;
+if (fullNum %2 ===0){
+    evenNum++;
+};
+console.log(evenNum);
+}
+while(sumIteration<=100);
+
+let evenNumbersCount = 0;
+for (let t=0; t<100; t ++);
+let ranNum= Math.floor(Math.random()*50)+1;
+console.log(ranNum);
+if(ranNum % 2 === 0){
+    evenNumbersCount++;
+}
+console.log(evenNumbersCount);
+  
+
+
+
+// 14. Find First Prime Number: Write a while loop that generates random numbers between 1 and 100 until it finds a prime number. Print the prime number (https://www.techtarget.com/whatis/definition/prime-number#:~:text=A%20prime%20number%20is%20a,%2C%2019%2C%2023%20and%2029.) and the number of iterations it took to find it.
+console.log("Exercise 14");
+let iterationNum = 0;
+let totIterations = 0;
+let primeNumber;
+do{
+    let isPrime = true;
+    let allNum =  Math.floor(Math.random()*100)+1; 
+    for (let i=2; i<= Math.sqrt(allNum); i++){
+        if (allNum %i === 0){
+            isPrime = false;
+            break;
+        }
+    }
+    if (isPrime && allNum>1){
+        primeNumber = allNum;
+        break;
+    }
+    iterationNum++;
+}
+while (totIterations <=100);
+console.log("iterations" + iterationNum);
+console.log("prime number"+ primeNumber);
+
+
+// 15. Accumulate Multiples of Three: Write a while loop that continuously generates random numbers between 1 and 15. Stop the loop once the sum of numbers that are multiples of three exceeds 50. Print the final sum and the total number of multiples of three.
+console.log("Exercise 15");
+let randNumber = 0;
+let totalMultiplesOfThree = 0; 
+let resultSum = 0;
+let m=1;
+while (resultSum<=50){
+    randNumber = Math.floor(Math.random()*15)+1;
```

Comment:
When you remove 'randNumber' variable declaration above, you'd have to change this line to 'const randNumber = Math.floor(Math.random()*15)+1;' (you can use const as you don't change the value of it but re-create the variable on each run of the loop